File: clean_admin.py
import re

with open('./Frontend/src/pages/AdminLogin.jsx', 'r', encoding='utf-8') as f:
    code = f.read()

# 1. Replace state
code = code.replace('const [role, setRole] = useState("FARMER_GROUP");', 'const role = "ADMIN";')

# 2. Remove SlideToggle import
code = code.replace('import SlideToggle from "../components/common/SlideToggle";\n', '')

# 3. Remove ROLE_OPTIONS
code = re.sub(r'const ROLE_OPTIONS = \[.*?\];\n\n', '', code, flags=re.DOTALL)

# 4. Remove Role toggle and Admin button UI
code = re.sub(r'\{\/\* Role toggle and Admin button \*\/\}[\s\S]*?\{\/\* Form \*\/\}', '{/* Form */}', code)

# 5. Simplify header and subtitle
# In Login.jsx, header is currently just "Welcome back" (my previous script fixed it for Login)
# Wait, let me check what's actually in AdminLogin.jsx (it's identical to Login.jsx right now)
code = re.sub(r'<h2 className="text-5xl font-extrabold text-white leading-14 mb-4">.*?<span className="text-blue-300">', '<h2 className="text-5xl font-extrabold text-white leading-14 mb-4">\n              The\n              <span className="text-violet-300">', code, flags=re.DOTALL)

# Change all emerald colors to violet
code = code.replace('emerald', 'violet')

# Change background blurs
code = code.replace('bg-white/5', 'bg-violet-400/5')
code = code.replace('bg-blue-400/10', 'bg-violet-400/10')

# Change title
code = code.replace('"Welcome back"', '<><Icon icon="ph:shield-check-fill" className="text-violet-500" /> Admin Login</>')

# Change subtitle
code = code.replace('"Sign in to your FarmFresh account"', '"Sign in to access the administrator portal"')

# Change email placeholder
code = code.replace('placeholder={role === "FARMER_GROUP" ? "farmer@farmfresh.com" : "collective@farmfresh.com"}', 'placeholder="admin@farmfresh.com"')

# Remove social login
# It looks like: {/* Social Login Separator */} ... </button>
code = re.sub(r'\{\/\* Social Login Separator \*\/\}[\s\S]*?<\/button>\s*</div>', '</div>', code)
code = re.sub(r'\{\/\* Social Login Separator \*\/\}[\s\S]*?(?:<\/button>)\n', '', code)

# Remove register link
code = re.sub(r'<p\s*className=\{\`text-center text-sm mt-5[\s\S]*?Don\'t have an account\?[\s\S]*?<\/p>', '', code)

# Remove remember me
code = re.sub(r'<label className="flex items-center gap-2 cursor-pointer">[\s\S]*?<\/label>', '', code)

# Remove the empty flex justify-between div around Remember Me / Forgot Password?
code = code.replace('<div\n                      className={`flex justify-between pb-1.5 ${isDark ? "text-slate-300" : "text-slate-600"}`}\n                    >\n\n                      <button', '<div\n                      className={`flex justify-end pb-1.5 ${isDark ? "text-slate-300" : "text-slate-600"}`}\n                    >\n                      <button')

code = code.replace('<div\n                      className={`flex justify-between pb-1.5 ${isDark ? "text-slate-300" : "text-slate-600"}`}\n                    >', '<div\n                      className={`flex justify-end pb-1.5 ${isDark ? "text-slate-300" : "text-slate-600"}`}\n                    >')


with open('./Frontend/src/pages/AdminLogin.jsx', 'w', encoding='utf-8') as f:
    f.write(code)

print('AdminLogin.jsx successfully styled to violet')

File: clean_admin.js
const fs = require('fs');
let code = fs.readFileSync('./Frontend/src/pages/AdminLogin.jsx', 'utf8');

// 1. Remove ROLE_OPTIONS
code = code.replace(/const ROLE_OPTIONS = \[[\s\S]*?\];/m, '');

// 2. Remove role toggle and Admin button
code = code.replace(/\{\/\* Role toggle and Admin button \*\/\}[\s\S]*?\{\/\* Form \*\/\}/m, '{/* Form */}');

// 3. Remove Social Login Separator and Google Login Button
code = code.replace(/\{\/\* Social Login Separator \*\/\}[\s\S]*?\{\/\* Google Login Button \*\/\}[\s\S]*?<\/button>\s*<\/>\s*\)}/m, '');

// 4. Remove Register link
code = code.replace(/<p\s*className=\{\	ext-center text-sm mt-5[\s\S]*?<\/p>/m, '');

// 5. Remove 'Remember me'
code = code.replace(/<label className="flex items-center gap-2 cursor-pointer">[\s\S]*?<\/label>/m, '');

// 6. Simplify Header
code = code.replace(/\{role === "ADMIN" \? \([\s\S]*?\) : \([\s\S]*?\)\}/m, '<><Icon icon="ph:shield-check-fill" className="text-violet-500" /> Admin Login</>');

code = code.replace(/\{role === "ADMIN"[\s\S]*?\? "Sign in to access the administrator portal"[\s\S]*?: "Sign in to your FarmFresh account"\}/m, '"Sign in to access the administrator portal"');

// 7. Simplify placeholders
code = code.replace(/placeholder=\{[\s\S]*?\}/m, 'placeholder="admin@farmfresh.com"');

// 8. Simplify button classes where role ternaries exist
code = code.replace(/role === "ADMIN"[\s\S]*?\? "bg-violet-600 border-violet-500 text-white shadow-lg shadow-violet-500\/20"[\s\S]*?: "bg-gradient-to-r from-emerald-500 to-emerald-600 hover:from-emerald-400 hover:to-emerald-500 shadow-emerald-500\/20"/m, '"bg-gradient-to-r from-violet-500 to-violet-600 hover:from-violet-400 hover:to-violet-500 shadow-violet-500/20"');

// Actually wait, let's just write to file and I will manually multi_replace what is left.
fs.writeFileSync('./Frontend/src/pages/AdminLogin.jsx', code);
console.log('Cleaned admin login');
